### PR TITLE
[Rule Tuning] Local Scheduled Task Commands

### LINKED ISSUE #998: [Rule Tuning] Local Scheduled Task Commands
## Description
As is, the `Local Scheduled Task Commands` rule is too noisy. It can be further refined to eliminate noise. 

## Original Rule
```
event.category:process and event.type:(start or process_started) and
  process.name:schtasks.exe and
  process.args:(-change or -create or -run or -s or /S or /change or /create or /run)
```

## Potential Update

```
  process where event.type == "start" and
    (process.name : "schtasks.exe" or process.pe.original_file_name : "schtasks.exe") and
    process.parent.name : ("cmd.exe", "wscript.exe", "rundll32.exe", "regsvr32.exe",
      "wmic.exe", "mshta.exe","powershell.exe", "pwsh.exe") and
    process.args : "/create" and
    process.args : ("*/RU*", "*/SC*", "*/TN*", "*/TR*", "*/F*") and 
    /* exclude SYSTEM SID - look for task creations by non-SYSTEM user */
    not user.id : "S-1-5-18"
```

### PR DESCRIPTION
**Pull request by bm11100**
## Issues
resolves #998 

## Summary
As is, the `Local Scheduled Task Commands` rule is too noisy. It can be further refined to eliminate noise. This PR updates the rule to look for scheduled task creation events spawning from various script interpreters by a non SYSTEM user.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
 timestamp_override = "event.ingested"
-type = "query"
+type = "eql"
 
 query = '''
-event.category:process and event.type:(start or process_started) and
-  process.name:schtasks.exe and
-  process.args:(-change or -create or -run or -s or /S or /change or /create or /run)
+process where event.type == "start" and
+  (process.name : "schtasks.exe" or process.pe.original_file_name : "schtasks.exe") and
```
Comment (brokensound77) [inline rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml:24]:
```suggestion
  (process.name : "schtasks.exe" or process.pe.original_file_name == "schtasks.exe") and
```

File: rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml [hunk 2]
```diff
@@ -1,28 +1,33 @@
...
+  process.parent.name : ("cmd.exe", "wscript.exe", "rundll32.exe", "regsvr32.exe",
+                         "wmic.exe", "mshta.exe","powershell.exe", "pwsh.exe") and
+  process.args : "/create" and
```
Comment (brokensound77) [inline rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml:27]:
```suggestion
  process.args : ("/create", "-create") and
```

File: rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml [hunk 3]
```diff
@@ -1,28 +1,33 @@
...
+  process.args : ("*/RU*", "*/SC*", "*/TN*", "*/TR*", "*/F*") and
```
Comment (brokensound77) [inline rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml:28]: I see these all parsed independently in sample data - any reason for the wildcards?

**Review by brokensound77 — APPROVED** [review]
Few comments then LGTM 👍

**Review by Samirbous — APPROVED** [review]
added few parent processes, otherwise LGTM

File: rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml [hunk 4]
```diff
@@ -1,28 +1,34 @@
...
 tags = ["Elastic", "Host", "Windows", "Threat Detection", "Persistence"]
 timestamp_override = "event.ingested"
-type = "query"
+type = "eql"
 
 query = '''
-event.category:process and event.type:(start or process_started) and
-  process.name:schtasks.exe and
-  process.args:(-change or -create or -run or -s or /S or /change or /create or /run)
+process where event.type == "start" and
```
Comment (Samirbous) [inline rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml:23]: @bm11100 reviewing a sample with @dstepanic17 we would recommend to include also unsigned parents (using a sequence), something like below:

`
sequence with maxspan=5m 
[process where event.type != "end" and 
  (process.name : ("cmd.exe", "wscript.exe", "rundll32.exe", "regsvr32.exe", "wmic.exe", "mshta.exe","powershell.exe", "pwsh.exe", "WmiPrvSe.exe", "wsmprovhost.exe", "winrshost.exe") or process.code_signature.trusted == false)] by process.entity_id
  [process where event.type == "start" and
  (process.name : "schtasks.exe" or process.pe.original_file_name == "schtasks.exe") and process.args : ("/create", "-create") and
  process.args : ("*/RU*", "*/SC*", "*/TN*", "*/TR*", "*/F*", "/XML") and
  /* exclude SYSTEM SID - look for task creations by non-SYSTEM user */
  not user.id : ("S-1-5-18", "S-1-5-19", "S-1-5-20")] by process.parent.entity_id
`

Comment (bm11100) [reply rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml]: updated to sequence, tested against sample and detected

File: rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml (hunk 3 above)
Comment (bm11100) [reply rules/windows/persistence_scheduled_task_creation_via_script_interpreter.toml]: eliminated the wildcards

File: rules/windows/persistence_local_scheduled_task_creation.toml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+name = "Scheduled Task/Job"
+reference = "https://attack.mitre.org/techniques/T1053/"
+
+[[rule.threat.technique.subtechnique]]
+id = "T1053.005"
+name = "Scheduled Task"
+reference = "https://attack.mitre.org/techniques/T1053/005/"
+
+
+
```
Comment (dstepanic17) [inline rules/windows/persistence_local_scheduled_task_creation.toml:49]: @bm11100 Extra line need to be removed?

**Comment by dstepanic17** [comment]
@bm11100 For the second sequence in the rule, we leveraged the original file name should we do that in the first sequence as well for each program?

File: rules/windows/persistence_local_scheduled_task_creation.toml (hunk 5 above)
Comment (bm11100) [reply rules/windows/persistence_local_scheduled_task_creation.toml:49]: the linter always does 3 new lines and a new line at the end. we can manually modify if we want, but shouldn't affect anything. eventually i think we will nest it  
```
[[rule.threat]]
framework = "MITRE ATT&CK"

  [rule.threat.tactic]
  name = "Execution"
  id = "TA0002"
  reference = "https://attack.mitre.org/tactics/TA0002/"

  [[rule.threat.technique]]
  id = "T1059"
  name = "Command and Scripting Interpreter"
  reference = "https://attack.mitre.org/techniques/T1059/"

    [[rule.threat.technique.subtechnique]]
    name = "Unix Shell"
    id = "T1059.004"
    reference = "https://attack.mitre.org/techniques/T1059/004/"
```

**Comment by bm11100** [comment]
> @bm11100 For the second sequence in the rule, we leveraged the original file name should we do that in the first sequence as well for each program?

yeah good call, i think we should use original file name when we can to account for renames, ill update it

**Review by dstepanic17 — APPROVED** [review]
LGTM

**Pull request merged**